rdt/transformers/datetime.py

```diff
@@ -117,8 +119,10 @@ def _fit(self, data):
             data (pandas.Series):
                 Data to fit the transformer to.
         """
-        if self.datetime_format is None and data.dtype == 'object':
-            self.datetime_format = _guess_datetime_format_for_array(data.to_numpy())
+        self._dtype = data.dtype
+        if self.datetime_format is None:
+            datetime_array = data.astype(str).to_numpy()
```

Comment:
This way we are able to properly capture the `str` representation of the `dataframe`.
If we have a representation of a `datetime[ns]` as `2022-04-05`, if we convert it directly to `numpy` it will be represented as `2022-04-05 00:00:00000000` which is not the intended output format.